src/main/scala/is/hail/table/Table.scala

```diff
@@ -417,6 +417,9 @@ class Table(val hc: HailContext, val tir: TableIR) {
             case (None, Some(y)) =>
               info(s"ROW IN RIGHT, NOT LEFT: ${ y.mkString("\n    ") }\n")
               false
+            case (None, None) =>
+              assert(false)
+              false
```

Comment:
ah right, `assert(false)` returns unit...

This is fine. Another option is to throw something: `throw new IllegalStateException()`, that never returns so the type checker is happy.